### DISCUSSION TITLE
Dynamically construct `jwk_set_url` by using `kid` as part of the URL

### DISCUSSION THREAD

**Discussion by anilmujagic** [grafana/grafana#44196] [General]
I'm using Grafana behind AWS load-balancer, which is responsible for authentication and is then providing a JWT in the request header to Grafana. However, I can't configure the JWT auth due to Grafana requiring statically configured `jwk_set_url`, which is not possible in this case, since AWS load-balancer has a different URL for each key, as explained in [the documentation](https://docs.aws.amazon.com/elasticloadbalancing/latest/application/listener-authenticate-users.html#user-claims-encoding).

Would it be possible to enable specifying a placeholder in the `jwk_set_url`, which will be replaced by `kid` or even any other part of the JWT. For example:

```ini
[auth.jwt]
enabled = true
header_name = x-amzn-oidc-data
username_claim = sub
email_claim = email
jwk_set_url = https://public-keys.auth.elb.eu-west-1.amazonaws.com/{{kid}}/.well-known/jwks.json
```

**Comment by komalda** [comment]
Any update on this? Did you able to find the way.

**Reply by anilmujagic** [reply]
Nope, but you're welcome to upvote it to get more visibility 
<img width="76" alt="image" src="https://user-images.githubusercontent.com/902307/185717650-53951e14-9768-4d24-bba2-201321dcf542.png">

**Comment by gelicia** [comment]
Hello, as you may have heard, we are [transitioning away from using discussions to discuss feature requests](https://github.com/grafana/grafana/discussions/73424). Due to the age and number of responses to this discussion, we are deciding to close it. If this is something you would like to see in Grafana, feel free to open an [issue](https://github.com/grafana/grafana/issues) so the discussion can continue. Thank you!

**Reply by clayrosenthal** [reply]
Opened an [issue](https://github.com/grafana/grafana/issues/109373)
